### DISCUSSION TITLE
proxy RTSP PTZ controls to v4l2-ctl?

### DISCUSSION THREAD

**Discussion by akohlsmith** [bluenviron/mediamtx#2921] [Questions]
### Question

I've got a nice Obsbot PTZ camera that works great with mediamtx+ffmpeg. I am trying to find a way to present the PTZ controls that this camera has to any RTSP clients which support PTZ controls, but I'm not sure how to do this?

Essentially I'd like to set up a control endpoint which can accept PTZ and other controls such as brightness/focus/sharpness/etc and translate these to individual `v4l2-ctl` commands. What I'm struggling with is the "so any PTZ-aware RTSP client can use" part.

How can I do this with mediamtx? If RTSP isn't the best method for this, is there an example using another transport?

**Comment by skaface** [comment]
This doesn't directly answer your question but it might be a suitable alternative solution. I use MPV to view the stream from mediamtx, and have a MPV script which forwards PTZ commands straight to the camera via HTTP POST commands allowing me to control the camera using key presses inside MPV

**Comment by aler9** [comment]
Hello, PTZ controls are not passed through RTSP, which is video-only, but are usually passed through the ONVIF protocol (which is based on SOAP, which is based on HTTP), which is currently outside the scope of MediaMTX.

My advice is to start interacting with ONVIF (there are lots of available clients), which is basically a buch of HTTP POST requests, and then, after you are sure that it works with your camera, you can set up a bridge from `v4l2-ctl` to ONVIF commands.

[a comment by an automated account is not shown]
